src/Imath/ImathFrustum.h

```diff
@@ -151,18 +152,32 @@ template <class T> class Frustum
     //----------------------------------------------------------
 
     IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Line3<T> projectScreenToRay (const Vec2<T>&) const;
-    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Vec2<T> projectPointToScreen (const Vec3<T>&) const;
-
-    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 T ZToDepth (long zval, long min, long max) const;
-    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 T normalizedZToDepth (T zval) const;
-    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 long DepthToZ (T depth, long zmin, long zmax) const;
-
-    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 T worldRadius (const Vec3<T>& p, T radius) const;
-    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 T screenRadius (const Vec3<T>& p, T radius) const;
+    IMATH_CONSTEXPR14 Vec2<T> projectPointToScreen (const Vec3<T>&) const;
+    IMATH_HOSTDEVICE IMATH_CONSTEXPR14 Vec2<T>
+    projectPointToScreen_noexcept (const Vec3<T>&) const noexcept;
```

Comment:
I think we should change the behavior of these but make the names consistent, such that projectPointToScreen becomes noexcept and projectPointToScreenExc gets added

Reply:
haha, what Cary and Larry said :)

Reply:
We could go one step further, and simply remove the throwing. Elsewhere we explicitly make promises that Foo will throw if it is named FooExc; we could argue that throwing was an implementation error. We could hold off on adding Exc methods until an application developer specifically asks us for it...